TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -389,6 +389,49 @@ public override bool Visit(DataDefinition dataDefinition)
 
             }
 
+            //Check if variable is Sync inside Strict Type definition
+            if (dataDefinition.SemanticData.HasFlag(Symbol.Flags.InsideTypedef) &&
+                dataDefinition.SemanticData.HasFlag(Symbol.Flags.Sync))
+            {
+                //Typedef instruction => check if it's marked Strict
+                if (dataDefinition.SemanticData.Kind == Symbol.Kinds.Typedef && dataDefinition.SemanticData.HasFlag(Symbol.Flags.Strict))
+                {
+                    DiagnosticUtils.AddError(dataDefinition, $"Cannot declare Type definition {dataDefinition.Name} with Sync clause because it is Strict.");
+                }
+                //Variable inside Typedef => check if parent typedef is marked Strict
+                else if (dataDefinition.SemanticData.NearestParent(Symbol.Kinds.Typedef).HasFlag(Symbol.Flags.Strict))
+                {
+                    DiagnosticUtils.AddError(dataDefinition, $"{dataDefinition.Name} cannot be declared with Sync clause because it is part of Strict Type definition {dataDefinition.ParentTypeDefinition?.Name}.");
+                }
+
+            }
+            
+            //Variable is of a user defined Strict Type
+            if (dataDefinition.SemanticData.HasFlag(Symbol.Flags.HasATypedefType) && dataDefinition.DataType.RestrictionLevel == RestrictionLevel.STRICT)
```

Comment:
DataType is the old type mechanism that must be replaced by new Type class.
New Type class doesn't have the resolved TypeDefinition yet.
You can use dataDefinition.TypeDefinition?.RestrictionLevel in the meantime.